envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/permissions/SourceIpBasedAuthenticationTest.kt

```diff
@@ -0,0 +1,123 @@
+
+internal class SourceIpBasedAuthenticationTest : EnvoyControlTestConfiguration() {
+
+    companion object {
+        @Container
+        val loremContainer = ToxiproxyContainer(exposedPortsCount = 1).withNetwork(network)
+        @Container
+        val ipsumContainer = ToxiproxyContainer(exposedPortsCount = 1).withNetwork(network)
+
+        private const val prefix = "envoy-control.envoy.snapshot"
+        private val properties = { mapOf(
+            "$prefix.incoming-permissions.enabled" to true,
+            "$prefix.outgoing-permissions.services-allowed-to-use-wildcard" to setOf("echo"),
+            "$prefix.incoming-permissions.source-ip-authentication.ip-from-service-discovery.enabled-for-incoming-services" to
+                    listOf("echo"),
+            "$prefix.incoming-permissions.source-ip-authentication.ip-from-range.lorem" to "${loremContainer.ipAddress()}/32",
+            "$prefix.routes.status.create-virtual-cluster" to true,
+            "$prefix.routes.status.path-prefix" to "/status/",
+            "$prefix.routes.status.enabled" to true
+        ) }
+
+        // language=yaml
+        private val echo2EnvoyConfig = Echo2EnvoyAuthConfig.copy(configOverride = """
+            node:
+              metadata:
+                proxy_settings:
+                  incoming:
+                    endpoints:
+                      - path: "/secured_endpoint"
+                        clients: ["echo", "lorem"]
+        """.trimIndent())
+
+        @JvmStatic
+        @BeforeAll
+        fun setupTest() {
+            setup(appFactoryForEc1 = { consulPort ->
+                EnvoyControlRunnerTestApp(properties = properties(), consulPort = consulPort)
+            }, envoyConfig = Echo1EnvoyAuthConfig, secondEnvoyConfig = echo2EnvoyConfig, envoys = 2)
+        }
+    }
+
+    @Test
+    fun `should allow access to selected clients using ip-from-discovery based authentication over plain http`() {
+        registerEcho1WithEnvoy1OnIngress()
+        registerEcho2WithEnvoy2OnIngress()
+
+        untilAsserted {
+            envoyContainer2.admin().resetCounters()
+
+            // when
+            val requestFromEcho1Response = callEcho2ThroughEnvoy1()
+            val directRequestResponse = callEcho2ThroughEnvoy2Ingress()
+
+            val plainHttpAccessDenials = envoyContainer2.admin().statValue("http.ingress_http.rbac.denied")?.toInt()
+            val sslHandshakes = envoyContainer2.admin().statValue("listener.0.0.0.0_5001.ssl.handshake")?.toInt()
+
+            // then
+            assertThat(requestFromEcho1Response).isOk().isFrom(echoContainer2)
+            assertThat(directRequestResponse).isForbidden()
+
+            assertThat(sslHandshakes).isZero()
+            assertThat(plainHttpAccessDenials).isOne()
+        }
+    }
+
+    @Test
+    fun `should allow access to selected clients using ip-from-range based authentication over plain http`() {
```

Comment:
This is new right?

Reply:
yes, I added a test for ip from static range because we didn't have any